### ISSUE TITLE
Add a setup.py file

### ISSUE THREAD

**Issue by timvieira** [jkkummerfeld/berkeley-parser-analyser#2]
Thanks for the great package. In an effort to make the package easier to use. I'd strongly recommend making it a python package by simply adding a ``setup.py`` file, which should (minimally) look like:

    from setuptools import setup
    setup(name='berkeley_parse_analyser',
          version='1.0',
          author='Jonathan Kummerfeld',
          description='A tool for classifying mistakes in the output of parsers',
          packages=['berkeley_parse_analyser'])

Note: It's conventional rename the ``src`` directory to the name of the project, e.g., ``berkeley_parse_analyser``.

It would also be a good idea to run this code through pylint and the pep8 coding style checker (which pylint will do for you).

**Comment by jkkummerfeld** [comment]
Thanks for the suggestion! I followed your suggestions to make this change, and started making fixes as recommended by pylint. There are many more to address (and in an ideal world I would also make this python 3 compatible), but that will have to wait for another day.

**Comment by timvieira** [comment]
Awesome! Thanks for replying so quickly :-D I'm glad you like the suggestions. Another one I'd recommend is using python classes instead of dicts for storing info, e.g. [here](https://github.com/jkkummerfeld/berkeley-parser-analyser/blob/master/berkeley_parse_analyser/transform_search.py#L61)

```python
def gen_missing_successor(ctree, error):
    # ...
    info = {
        'type': 'add',
        'label': get_label(nnode),
        'span': nnode.span,
        'subtrees': [get_label(subtree) for subtree in nnode.subtrees],
        'parent': nnode.parent.label,
        # etc...
    }
    # ...
```

Becomes:
```python
class Transform:
    def __init__(self, nnode, ttype)
        self.type = ttype
        self.label = get_label(nnode)
        self.span = nnode.span
        self.subtrees = [get_label(subtree) for subtree in nnode.subtrees]
        self.parent = nnode.parent.label
        # etc...

def gen_missing_successor(ctree, error):
    # ...
    info = Transform(nnode, ttype='add')
    # ...
```

**Comment by jkkummerfeld** [comment]
Yeah... that was definitely a case of a quick and dirty solution that grew and grew.

**Comment by timvieira** [comment]
Ha! I know that phenomena all too well. Using dictionaries that way is very "Perl style" (as is quick 'n dirty). I thought maybe you were be a Perl hacker writing Python.

**Comment by jkkummerfeld** [comment]
I've written practically no Perl actually, but with sufficiently little effort poor programming style can be achieved in any language :)
